### PR TITLE
Apply Doctrine CS 4.0 + test for removeEventSubscriber() & unknown event dispatching

### PR DESCRIPTION
**Pull request by Majkl578**

### CODE REVIEW

File: lib/Doctrine/Common/EventManager.php [hunk 1]
```diff
@@ -1,45 +1,43 @@
...
      *
      * @return void
      */
-    public function dispatchEvent($eventName, EventArgs $eventArgs = null)
+    public function dispatchEvent($eventName, ?EventArgs $eventArgs = null)
     {
-        if (isset($this->_listeners[$eventName])) {
-            $eventArgs = $eventArgs === null ? EventArgs::getEmptyInstance() : $eventArgs;
+        if (! isset($this->_listeners[$eventName])) {
+            return;
```
Comment (Majkl578) [inline lib/Doctrine/Common/EventManager.php]: Any proposal how to cover this line by a test that would actually assert something?

Comment (malarzm) [reply lib/Doctrine/Common/EventManager.php]: Not sure if it'll count, but have a listener for `foo` that expects to be not called and dispatch `bar` event?

File: lib/Doctrine/Common/EventManager.php [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
+        }
+
+        $eventArgs = $eventArgs === null ? EventArgs::getEmptyInstance() : $eventArgs;
```
Comment (malarzm) [inline lib/Doctrine/Common/EventManager.php]: While I'm here, should this line be changed to use `??`

File: lib/Doctrine/Common/EventManager.php (hunk 1 above)
Comment (Majkl578) [reply lib/Doctrine/Common/EventManager.php]: That would indeed cover the line (also not registering anything would work), but sounds like cheating. :trollface: 
I got only one dirty idea: Use reflection to assert that EventArgs::$_emptyEventArgsInstance is `null` after dispatching an unknown event.

**Pull request merged**